src/NuGet.Core/NuGet.Common/NuGetEnvironment.cs

```diff
@@ -193,6 +196,57 @@ private static string GetHome()
             }
         }
 
+        public static string GetDotNetLocation()
+        {
+            string path = Environment.GetEnvironmentVariable("PATH");
+            foreach (var dir in path.Split(':', ';'))
+            {
+                string fullPathExe = Path.Combine(dir, DotNetExe);
+                if (File.Exists(fullPathExe))
+                {
+                    return fullPathExe;
+                }
+
+                string fullPath = Path.Combine(dir, DotNet);
```

Comment:
Can you only check for "dotnet" on non-Windows and "dotnet.exe" on Windows? No need to check for both.